docs/ossm/ossm-2-and-ossm-3-side-by-side/README.md

```diff
@@ -0,0 +1,98 @@
+    mysqldb-v1-7c4c44b9b4-22b57.bookinfo          Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-install-istio-system-bd58bdcd5-2htkf     1.20.8
+    productpage-v1-6f9c6589cb-l6rvg.bookinfo      Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-install-istio-system-bd58bdcd5-2htkf     1.20.8
+    ratings-v1-559b64556-f6b4l.bookinfo           Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-install-istio-system-bd58bdcd5-2htkf     1.20.8
+    ratings-v2-8ddc4d65c-bztrg.bookinfo           Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-install-istio-system-bd58bdcd5-2htkf     1.20.8
+    ratings-v2-mysql-cbc957476-m5j7w.bookinfo     Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-install-istio-system-bd58bdcd5-2htkf     1.20.8
+    reviews-v1-847fb7c54d-7dwt7.bookinfo          Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-install-istio-system-bd58bdcd5-2htkf     1.20.8
+    reviews-v2-5c7ff5b77b-5bpc4.bookinfo          Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-install-istio-system-bd58bdcd5-2htkf     1.20.8
+    reviews-v3-5c5d764c9b-mk8vn.bookinfo          Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-install-istio-system-bd58bdcd5-2htkf     1.20.8
+    ```
+    ```sh
+    $ istioctl ps -i istio-system3
+    NAME                                          CLUSTER        CDS                LDS                EDS                RDS                ECDS        ISTIOD                            VERSION
+    details-v1-57f6466bdc-5krth.bookinfo2         Kubernetes     SYNCED (2m40s)     SYNCED (2m40s)     SYNCED (2m34s)     SYNCED (2m40s)     IGNORED     istiod-ossm3-5b46b6b8cb-gbjx6     1.23.0
+    productpage-v1-5b84ccdddf-f8d9t.bookinfo2     Kubernetes     SYNCED (2m39s)     SYNCED (2m39s)     SYNCED (2m34s)     SYNCED (2m39s)     IGNORED     istiod-ossm3-5b46b6b8cb-gbjx6     1.23.0
+    ratings-v1-fb764cb99-kx2dr.bookinfo2          Kubernetes     SYNCED (2m40s)     SYNCED (2m40s)     SYNCED (2m34s)     SYNCED (2m40s)     IGNORED     istiod-ossm3-5b46b6b8cb-gbjx6     1.23.0
+    reviews-v1-8bd5549cf-xqqmd.bookinfo2          Kubernetes     SYNCED (2m40s)     SYNCED (2m40s)     SYNCED (2m34s)     SYNCED (2m40s)     IGNORED     istiod-ossm3-5b46b6b8cb-gbjx6     1.23.0
+    reviews-v2-7f7cc8bf5c-5rvln.bookinfo2         Kubernetes     SYNCED (2m40s)     SYNCED (2m40s)     SYNCED (2m34s)     SYNCED (2m40s)     IGNORED     istiod-ossm3-5b46b6b8cb-gbjx6     1.23.0
+    reviews-v3-84f674b88c-ftcqg.bookinfo2         Kubernetes     SYNCED (2m40s)     SYNCED (2m40s)     SYNCED (2m34s)     SYNCED (2m40s)     IGNORED     istiod-ossm3-5b46b6b8cb-gbjx6     1.23.0
+    ```
```

Comment:
Hey @FilipB ,

I want to make sure I understand this correctly.

As a OSSM 2.x user I need to:

1. Be running OSSM 2.6

If I am not on OSSM 2.6, then I need to upgrade to 2.6.

As a user with the Multitenant (default) deployment model, I need to:

1. Keep my OSSM v2.6 setup as is.
3. Install `Red Hat OpenShift Service Mesh 3` operator
4. Create `IstioCNI` resource in `istio-cni` namespace
5. Create `Istio` resource in `istio-system3`
6. Deploy your workloads and label the namespaces with `istio.io/rev=ossm3` label
7. Run $ oc `istioctl ps` to confirm the application workloads are manage by their respective control planes.

As a user with Cluster-Wide (Single Tenant) mesh deployment model, I need to:

1. Configure discovery selectors in my OSSM v2.6 SMCP CR
2. Install `Red Hat OpenShift Service Mesh 3` operator
3. Create `IstioCNI` resource in `istio-cni` namespace
4. Create `Istio` resource in `istio-system3`
5. Deploy your workloads and label the namespaces with `istio.io/rev=ossm3` label
6. Run $ oc `istioctl ps` to confirm the application workloads are manage by their respective control planes.

Reply:
Yes, we should probably instruct users to upgrade to 2.6 before trying this. This procedure is not tested with OSSM < 2.6.

The steps are correct, yes. "Create Istio resource in istio-system3" step is also configuring 'discoverySelectors' which is a crucial part.